src/modules/drop-erc-721.ts

```diff
@@ -56,7 +75,305 @@ export class DropErc721Module {
+    return await this.contractWrapper.readContract.ownerOf(tokenId);
+  }
+
+  // TODO extract all the claim condition logic to its own class
+
+  /**
+   * Get the currently active claim condition in this Drop
+   *
+   * @returns the claim condition metadata
+   */
+  public async getActiveClaimCondition(): Promise<ClaimCondition> {
+    const index =
+      await this.contractWrapper.readContract.getIndexOfActiveCondition();
+    const mc = await this.contractWrapper.readContract.getClaimConditionAtIndex(
+      index,
+    );
+    return await this.transformResultToClaimCondition(mc);
+  }
+
+  /**
+   * Get all the claim conditions associated with this Drop
+   *
+   * @returns the claim conditions metadata
+   */
+  public async getAllClaimConditions(): Promise<ClaimCondition[]> {
+    const claimCondition =
+      await this.contractWrapper.readContract.claimConditions();
+    const count = claimCondition.totalConditionCount.toNumber();
+    const conditions = [];
+    for (let i = 0; i < count; i++) {
+      conditions.push(
+        await this.contractWrapper.readContract.getClaimConditionAtIndex(i),
+      );
+    }
+    return Promise.all(
+      conditions.map((c) => this.transformResultToClaimCondition(c)),
+    );
+  }
+
+  /**
+   * Get the total supply for this Drop.
+   *
+   * @returns the total supply
+   */
+  public async totalSupply(): Promise<BigNumber> {
+    return await this.contractWrapper.readContract.nextTokenIdToMint();
+  }
+
+  /**
+   * Get the unclaimed supply for this Drop.
+   *
+   * @returns the unclaimed supply
+   */
+  public async totalUnclaimedSupply(): Promise<BigNumber> {
+    return (await this.contractWrapper.readContract.nextTokenIdToMint()).sub(
+      await this.totalClaimedSupply(),
+    );
+  }
+
+  /**
+   * Get the claimed supply for this Drop.
+   *
+   * @returns the claimed supply
+   */
+  public async totalClaimedSupply(): Promise<BigNumber> {
+    return await this.contractWrapper.readContract.nextTokenIdToClaim();
+  }
+
+  /**
+   * Get NFT Balance
+   *
+   * @remarks Get a wallets NFT balance (number of NFTs in this module owned by the wallet).
+   *
+   * @example
+   * ```javascript
+   * // Address of the wallet to check NFT balance
+   * const address = "{{wallet_address}}";
+   *
+   * const balance = await module.balanceOf(address);
+   * console.log(balance);
+   * ```
+   */
+  public async balanceOf(address: string): Promise<BigNumber> {
+    return await this.contractWrapper.readContract.balanceOf(address);
+  }
+
+  /**
+   * Get NFT Balance for the currently connected wallet
+   */
+  public async balance(): Promise<BigNumber> {
+    return await this.balanceOf(await this.contractWrapper.getSignerAddress());
+  }
+
+  /**
+   * Get whether this wallet has approved transfers from the given operator
+   * @param address - the wallet address
+   * @param operator - the operator address
+   */
+  public async isApproved(address: string, operator: string): Promise<boolean> {
```

Comment:
how come we pass operator here? shouldn't it always be the drop contract address?